drivers/serial/Kconfig

```diff
@@ -31,6 +31,13 @@ config SERIAL_SUPPORT_INTERRUPT
 	  This is an option to be enabled by individual serial driver
 	  to signal that the driver and hardware supports interrupts.
 
+config SERIAL_SUPPORT_DATA_LINE_INVERSION
```

Comment:
Open question: Is this symbol actually required?

Reply:
Added this symbol to make sure that the vendors where pin inversion is not supported uart config structure will remain the same. But I'm open if we wan to include tx/rx invert property irrespective of hardware support

This implementation takes reference from CAN_FD_MODE property in CAN

Reply:
Up to @dcpleung to settle on this.

Reply:
Since this feature is not universal, I don't think it is a good idea to add it in the common `uart_config` struct. I suggest that for those drivers that support this feature, create driver specific config structs for them. These structs can have a common `uart_config` as the first element in there. So you can type-cast it both way depending on usage.